components/calendar/calendar.js

```diff
@@ -325,57 +331,66 @@ class Calendar extends LocalizeStaticMixin(LitElement) {
 
 		const dates = getDatesInMonthArray(this._shownMonth, this._shownYear);
 		const dayRows = dates.map((week) => {
-			const weekHtml = week.map((day) => {
+			const weekHtml = week.map((day, index) => {
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				const selected = this.selectedValue ? checkIfDatesEqual(day, parseISODate(this.selectedValue)) : false;
 				const classes = {
 					'd2l-calendar-date': true,
-					'd2l-calendar-date-selected': this.selectedValue ? checkIfDatesEqual(day, parseISODate(this.selectedValue)) : false,
+					'd2l-calendar-date-selected': selected,
 					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
 				};
-				const focused = checkIfDatesEqual(day, this._focusDate);
+				const year = day.getFullYear();
+				const month = day.getMonth();
+				const date = day.getDate();
 				return html`
 					<td>
 						<div
+							aria-label="${formatDate(day, {format: 'full'})}"
+							aria-selected="${selected}"
 							@click="${this._onDateSelected}"
 							@keydown="${this._onKeyDown}"
 							class=${classMap(classes)}
-							data-date=${day.getDate()}
-							data-month=${day.getMonth()}
-							data-year=${day.getFullYear()}
+							data-date=${date}
+							data-month=${month}
+							data-year=${year}
+							headers="${this._tableInfoId}-weekday-${index}"
+							id="${this._tableInfoId}-${year}-${month}-${date}"
+							role="gridcell"
 							tabindex=${focused ? '0' : '-1'}>
-							${day.getDate()}
+							${date}
 						</div>
 					</td>`;
 			});
 
-			return html`<tr>${weekHtml}</tr>`;
+			return html`<tr role="row">${weekHtml}</tr>`;
 		});
+		const activeDate = `${this._tableInfoId}-${this._focusDate.getFullYear()}-${this._focusDate.getMonth()}-${this._focusDate.getDate()}`;
 		const heading = `${calendarData.descriptor.calendar.months.long[this._shownMonth]} ${this._shownYear}`;
 		return html`
-			<div aria-labelledby="${this._labelId}" class="d2l-calendar">
-				<table summary="${this.summary}" role="grid">
-					<thead>
-						<tr class="d2l-calendar-title">
-							<td>
-								<d2l-button-icon
-									@click="${this._onPrevMonthButtonClick}"
-									text="${this._computeText(getPrevMonth(this._shownMonth))}"
-									icon="tier1:chevron-left">
-								</d2l-button-icon>
-							</td>
-							<th colspan="5">
-								<h2 class="d2l-heading-4" aria-live="polite" id="${this._labelId}">${heading}</h2>
-							</th>
-							<td>
-								<d2l-button-icon
-									@click="${this._onNextMonthButtonClick}"
-									text="${this._computeText(getNextMonth(this._shownMonth))}"
-									icon="tier1:chevron-right">
-								</d2l-button-icon>
-							</td>
-						</tr>
-						<tr>${weekdayHeaders}</tr>
+			<div aria-labelledby="${this._tableInfoId}-heading" class="d2l-calendar">
```

Comment:
I'm curious... since this `<div>` doesn't have a role or anything, does the label actually do anything?

Reply:
ah nope I don't think it is needed. I was debating if this one was needed. It looks like there might be some form of this needed when `calendar` is in `dropdown` but I'll take a look more at that in that context.